.editorconfig

```diff
@@ -0,0 +1,13 @@
+root = true
+
+[*]
+indent_style = space
+indent_size = 2
```

Comment:
I think we want `4`.
Only in the case of unordered/ordered lists do we want `2`..

Reply:
If we do that we'll need to add a block for yaml/yml files as the indentation requirement there is 2 spaces no tabs:
https://docs.saltproject.io/en/latest/topics/troubleshooting/yaml_idiosyncrasies.html

As well as consider if shell needs to have a different indent style/setting as well. Part of why i just ran with these defaults is it covers a lot of this "for free". Let me see what the shell looks like change wise with 4 spaces. I think it'll change a lot more loc.

Reply:
A quick and very rough pass on 4 spaces for sh vs 2 (just looking for hunk +/1 lines it will catch a bit more than that though so take with salt):

4 spaces:
```
$ act -j sh-checker stdboth | grep -Ec '\|\ [-+]'
949
zsh: exit 1     act -j sh-checker 2>&1 | 
zsh: done       grep -Ec '\|\ [-+]'
```

2 spaces
```
$ act -j sh-checker stdboth | grep -Ec '\|\ [-+]' 
461
zsh: exit 1     act -j sh-checker 2>&1 | 
zsh: done       grep -Ec '\|\ [-+]'
```

I didn't do a comprehensive check but tabs is even more changes as it touches every indentation line.

So if we define this or want to define it we need to define:

Default for everything (my thought here is this should be the thing that most stuff needs)
Default for specific file types if it doesn't fit the above. Aka for adoc maybe we want tabs and 8 spaces, we would define a block for [*.adoc]

I didn't look for Makefiles but we may want a block for those too to specifiy tab and whatever tab stop we want for them

Reply:
Sorry for the delay.

I think we can ignore adoc for now.

Makefiles could use tabs.

Am I right in understanding that we're okay with using 4 spaces as a default?